### PR TITLE
ES backended filer support kv ops.

### PR DESCRIPTION
**Pull request by taozix**

### CODE REVIEW

**Comment by chrislusf** [comment]
Is there any way to tell Elastic not to index the content?

https://www.elastic.co/guide/en/elasticsearch/reference/current/enabled.html#enabled

**Comment by chrislusf** [comment]
separate into an `elastic_store_kv.go` file

File: weed/filer/elastic/v7/elastic_store.go [hunk 1]
```diff
@@ -66,15 +72,66 @@ func (store *ElasticStore) CommitTransaction(ctx context.Context) error {
...
 }
+
 func (store *ElasticStore) KvDelete(ctx context.Context, key []byte) (err error) {
-	return filer.ErrKvNotImplemented
+	id := fmt.Sprintf("%x", md5.Sum(key))
+	deleteResult, err := store.client.Delete().
+		Index(indexKV).
+		Type(indexType).
+		Id(id).
+		Do(context.Background())
```
Comment (chrislusf) [inline weed/filer/elastic/v7/elastic_store.go]: there is already a `ctx` parameter. No need to use `context.Background()`. Same for other places.

File: weed/filer/elastic/v7/elastic_store.go [hunk 2]
```diff
@@ -66,15 +72,66 @@ func (store *ElasticStore) CommitTransaction(ctx context.Context) error {
...
 	return nil
 }
+
 func (store *ElasticStore) KvDelete(ctx context.Context, key []byte) (err error) {
-	return filer.ErrKvNotImplemented
+	id := fmt.Sprintf("%x", md5.Sum(key))
+	deleteResult, err := store.client.Delete().
+		Index(indexKV).
+		Type(indexType).
+		Id(id).
```
Comment (chrislusf) [inline weed/filer/elastic/v7/elastic_store.go]: why not use `key` here?

Comment (taozix) [reply weed/filer/elastic/v7/elastic_store.go]: will cause problem if key contains '/'.

File: weed/filer/elastic/v7/elastic_store.go (hunk 1 above)
Comment (taozix) [reply weed/filer/elastic/v7/elastic_store.go]: it can be without indexing if totally no need to query.

Comment (chrislusf) [reply weed/filer/elastic/v7/elastic_store.go]: Only need to fetch the value by the key.

File: weed/filer/elastic/v7/elastic_store.go (hunk 2 above)
Comment (chrislusf) [reply weed/filer/elastic/v7/elastic_store.go]: no need to worry about that.

**Pull request merged**